packages/orm/src/BaseEntity.ts

```diff
@@ -21,9 +21,13 @@ export abstract class BaseEntity implements Entity {
   abstract id: string | undefined;
   readonly __orm: EntityOrmField;
 
-  protected constructor(em: EntityManager, metadata: any, data?: Record<string, any>) {
-    this.__orm = { em, metadata, data: data || {}, originalData: {} };
-    em.register(this);
+  protected constructor(em: EntityManager, metadata: any, defaultValues: object, opts: any) {
+    this.__orm = { em, metadata, data: { ...defaultValues }, originalData: {} };
+    // Ensure we have at least id set so the `EntityManager.register` works
+    if (typeof opts === "string") {
```

Comment:
when is opts a string?

Reply:
oh i see, with the new code path via hydrate.   Not sure I'm a fan of hidden behavior based on the type of an arg, although i suppose it's not publicly exposed since we don't intend for constructors to be directly called or overridden.

Reply:
Yeah, this went from "I'm just hacking this to see if it works" to "...oh huh it does..." to "...well, it's a private api so I guess...".

Agreed it should be better/more explicit eventually, but going to punt for now.